lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php

```diff
@@ -139,4 +190,39 @@ public function getConfigFE( \Aimeos\MShop\Order\Item\Base\Iface $basket )
 
 		return array_merge( $this->getProvider()->getConfigFE( $basket ), $this->getConfigItems( $feconfig ) );
 	}
+
+
+	/**
+	 * Sets the delivery attributes in the given service.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Base\Service\Iface $orderServiceItem Order service item that will be added to the basket
+	 * @param array $attributes Attribute key/value pairs entered by the customer during the checkout process
+	 */
+	public function setConfigFE( \Aimeos\MShop\Order\Item\Base\Service\Iface $orderServiceItem, array $attributes ) 
+	{
+		if( ( $code = $attributes['supplier.code'] ) != '' ) {
+			
+			if( $this->getConfigValue('supplier.display-name')
+				&& isset( $this->feConfig['supplier.code']['name'][$code] ) 
+			) {
+				// add name as attribute for summary page / customer email
+				$attributes['supplier.name'] = $this->feConfig['supplier.code']['name'][$code];
+			}
+			
+			if( $this->getConfigValue('supplier.display-address') 
+				&& isset( $this->feConfig['supplier.code']['short'][$code] ) 
+			) {
+				// add short address as attribute for summary page / customer email
+				$attributes['supplier.address'] = $this->feConfig['supplier.code']['short'][$code];
+			}
+			
+			if( $this->getConfigValue('supplier.display-code') === 0 ) {
+				// remove code attribute for summary page / customer email
+				unset( $attributes['supplier.code'] );
+			}
+			
+		}
+
+		$this->setAttributes( $orderServiceItem, $attributes, 'delivery' );
```

Comment:
This is wrong, you prevent all further decorators to modify the service attributes as well. Instead, you need to add your attributes (or remove existing ones) and call the next decorator using `$this->getProvider()->setConfigFE(...)`

Reply:
Sorry, didn't know that. I fixed it!